pkg/kubernetes/client.js

```diff
@@ -680,6 +697,34 @@ define([
         });
 
         /*
+         * add_watcher:
+         * @type: The type of watch to add
+         *
+         * Returns a Promise that if successfull will
+         * be a KubernetesWatch object for the given type.
+         * The same object will be available at self.watches[type]
+         */
+        self.add_watcher = function add_watcher (type) {
```

Comment:
The reason i based it off type is so that the attribute name in watcher is the same as what the caller passed in Otherwise it might not be obvious what the correct attribute is. I guess it's all documented in the API docs so maybe it doesn't matter much.